curvefs/src/metaserver/heartbeat.cpp

```diff
@@ -236,6 +239,30 @@ bool Heartbeat::GetProcMemory(uint64_t* vmRSS) {
     return false;
 }
 
+bool Heartbeat::GetMetaserverSpaceStatus(MetaServerSpaceStatus* status,
+                                         uint64_t ncopysets) {
+    StorageStatistics statistics;
+    auto kvStorage = GetStorageInstance();
+    bool succ = kvStorage->GetStatistics(&statistics);
+    if (!succ) {
+        LOG(ERROR) << "Get storage statistics failed";
+        return false;
+    }
+
+    status->set_memorythresholdbyte(statistics.MaxMemoryBytes);
+    status->set_memoryusedbyte(statistics.MemoryUsageBytes);
+    status->set_diskthresholdbyte(statistics.MaxDiskQuotaBytes);
+    status->set_diskusedbyte(0);
+    status->set_diskcopysetminrequirebyte(0);
+    if (ncopysets == 0) {
+        status->set_memorycopysetminrequirebyte(0);
+    } else {
+        status->set_memorycopysetminrequirebyte(
+            uint64_t(statistics.MemoryUsageBytes / ncopysets));
```

Comment:
Why do you calculate it this way

Reply:
> Why do you calculate it this way

It's a simplified strategy, we can balance metaserver storage in cluster according to its guidance.